Model/ResourceModel/NodeType/Product.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Snowdog\Menu\Model\ResourceModel\NodeType;
+
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Store\Model\Store;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product as CoreProduct;
+use Magento\Framework\EntityManager\MetadataPool;
+
+class Product extends AbstractNode
+{
+    /**
+     * @var MetadataPool
+     */
+    private $metadataPool;
+
+    public function __construct(
+        ResourceConnection $resource,
+        MetadataPool $metadataPool
+    ) {
+        $this->metadataPool = $metadataPool;
+        parent::__construct($resource);
+    }
+
+    /**
+     * @return array
+     * @throws \Exception
+     */
+    public function fetchConfigData()
+    {
+        $metadata = $this->metadataPool->getMetadata(ProductInterface::class);
+        $identifierField = $metadata->getIdentifierField();
+        $connection = $this->getConnection('read');
+
+        $select = $connection->select()->from(
+            ['e' => $this->getTable('catalog_product_entity')],
+            [$identifierField, 'sku']
```

Comment:
We can have many records with the same identifier (entity_id) in case of Magento Commerce.

Reply:
I think if we use one of my idea from prev comment this query will be unnecessary

Reply:
autocomplete functionality removed